debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/events/BufferingChangeStreamCursor.java

```diff
@@ -367,6 +391,27 @@ private ResumableChangeStreamEvent<TResult> pollWithDelay() {
         return event;
     }
 
+    public void pause() {
+        lock.lock();
+        try {
+            paused = true;
+        }
+        finally {
+            lock.unlock();
+        }
+    }
```

Comment:
This only synchronises the access to `paused`. It also means that when `condition#await` is called the current thread doesn't hold the lock